script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -84,34 +85,43 @@ class PubspecCheckCommand extends PackageLoopingCommand {
     }
 
     final List<String> pubspecLines = contents.split('\n');
-    final List<String> sectionOrder = pubspecLines.contains('  plugin:')
-        ? _majorPluginSections
-        : _majorPackageSections;
+    final bool isPlugin = pubspec.flutter?.containsKey('plugin') ?? false;
+    final List<String> sectionOrder =
+        isPlugin ? _majorPluginSections : _majorPackageSections;
     bool passing = _checkSectionOrder(pubspecLines, sectionOrder);
     if (!passing) {
-      print('${indentation}Major sections should follow standard '
+      printError('${indentation}Major sections should follow standard '
```

Comment:
Minor opportunistic cleanup; I think `printError` didn't exist yet when I wrote this code.